plugin/trino-spooling-filesystem/src/main/java/io/trino/spooling/filesystem/encryption/AzureSSEHeaders.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spooling.filesystem.encryption;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.filesystem.encryption.EncryptionKey;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.trino.spooling.filesystem.encryption.HeaderUtils.getOnlyHeader;
+
+public class AzureSSEHeaders
+        implements SSEHeaders
```

Comment:
Nit. Can you rename `*Header` to `*HeaderTranslator`. 
We try to reserve naming pattern `<SomeNoun>s` helper classes with static methods.

Reply:
Sure. I've changed that to EncryptionHeadersTranslator